### PR TITLE
SNOW-802269 Add ordering and size function for scala and java modules

### PR DESCRIPTION
**Pull request by sfc-gh-gmahadevan**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [X] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

 Adding missing ordering and size function available in python module but not in scala and java

## Pre-review checklist

(For Snowflake employees)

- [x] This change has passed precommit
- [ ] I have reviewed code coverage report for my PR in  ([Sonarqube](https://sonarqube.int.snowflakecomputing.com/dashboard?id=snowpark))

### CODE REVIEW

File: src/main/java/com/snowflake/snowpark_java/Functions.java [hunk 1]
```diff
@@ -3880,6 +3881,42 @@ public static Column listagg(Column col) {
     return new Column(com.snowflake.snowpark.functions.listagg(col.toScalaColumn()));
   }
 
+  /**
+   * Function to convert column name into column and order in descending manner.
```
Comment (sfc-gh-mrojas) [inline src/main/java/com/snowflake/snowpark_java/Functions.java:3885]: maybe use the same descriptin as here:
https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.functions.desc
Also you can add an example

File: src/main/java/com/snowflake/snowpark_java/Functions.java [hunk 2]
```diff
@@ -3880,6 +3881,42 @@ public static Column listagg(Column col) {
...
+   * @since 1.14.0
+   * @param name The input column name
+   * @return Column object ordered in descending manner.
+   */
+  public static Column desc(String name) {
+    return new Column(functions.desc(name));
+  }
+
+  /**
+   * Function to convert column name into column and order in ascending manner.
```
Comment (sfc-gh-mrojas) [inline src/main/java/com/snowflake/snowpark_java/Functions.java:3896]: Same as above but using https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.functions.asc

File: src/main/java/com/snowflake/snowpark_java/Functions.java [hunk 3]
```diff
@@ -3880,6 +3881,42 @@ public static Column listagg(Column col) {
...
+   * @since 1.14.0
+   * @param name The input column name
+   * @return Column object ordered in ascending manner.
+   */
+  public static Column asc(String name) {
+    return new Column(functions.asc(name));
+  }
+
+  /**
+   * Returns the size of the input ARRAY.
```
Comment (sfc-gh-mrojas) [inline src/main/java/com/snowflake/snowpark_java/Functions.java:3907]: check if this is consistent with: https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/latest/snowpark/api/snowflake.snowpark.functions.array_size

Also make sure the function is exposed both as array_size as well as size

File: src/main/java/com/snowflake/snowpark_java/Functions.java (hunk 2 above)
Comment (sfc-gh-gmahadevan) [reply src/main/java/com/snowflake/snowpark_java/Functions.java:3896]: addressed

File: src/main/java/com/snowflake/snowpark_java/Functions.java (hunk 3 above)
Comment (sfc-gh-gmahadevan) [reply src/main/java/com/snowflake/snowpark_java/Functions.java:3907]: yes, both size and array_size are present

**Comment by sfc-gh-bli** [comment]
@sfc-gh-gmahadevan 
1, do you have jira ticket for this PR? if yes, please add the JIRA number in the PR title, for example, `SNOW-XXXXX xxxxx`
2, do not push from your forked repo, you can create a branch in this repo and create PRs from this repo, otherwise, the merging gates don't work.

**Pull request merged**